src/script/lua_api/l_mainmenu.cpp

```diff
@@ -577,6 +577,13 @@ int ModApiMainMenu::l_get_favorites(lua_State *L)
 			lua_settable(L, top_lvl2);
 		}
 
+		float ping = servers[i]["ping"].asFloat();
+		if (ping) {
```

Comment:
asFloat() needs fixing, yes. That function (asFloat()) should not drop off the end of the function without a return IMO